toolchain/toolchain.bzl

```diff
@@ -54,11 +53,24 @@ hosts = {
     },
 }
 
-def _arm_gnu_toolchain(name, toolchain = "", toolchain_prefix = "",
-                       gcc_tool = "gcc", abi_version = "",
-                       target_compatible_with = [], copts = [],
-                       linkopts = [], version = "", include_std = False):
+def _arm_gnu_toolchain(
+        name,
+        toolchain = "",
+        toolchain_prefix = "",
+        gcc_tool = "gcc",
+        abi_version = "",
+        target_compatible_with = [],
+        copts = [],
+        linkopts = [],
+        version = "",
+        include_std = False):
     for host, exec_compatible_with in hosts[toolchain_prefix].items():
+        fix_linkopts = []
+
+        # macOS on apple silicon rejects the relative path LTO plugin
+        if version == "13.2.1" and host == "darwin_arm64":
+            fix_linkopts.append("-fno-lto")
```

Comment:
@AmateurECE, thanks for upgrading to `13.2.1`, however I have to disable the default enabled LTO otherwise I see this error:

```bash
error loading plugin: dlopen(external/_main~arm_toolchain~arm_none_eabi_darwin_arm64/bin/../libexec/gcc/arm-none-eabi/13.2.1/liblto_plugin.so, 0x0002): tried: 'external/_main~arm_toolchain~arm_none_eabi_darwin_arm64/bin/../libexec/gcc/arm-none-eabi/13.2.1/liblto_plugin.so' (relative path not allowed in hardened program)
```

I think this is caused by the higher security features on apple silicon.

Are we ok with this workaround for now?

Reply:
I'm totally fine with this! Thanks for catching this, I saw this in one of the CI runs, but it went away afterwards, so I thought I had inadvertently fixed it. I don't have a macOS machine handy, so I didn't test with it natively.

I should probably look to open an issue with ARM on their forums about it.